app/src/main/java/com/sun/moviedb_53/ui/detail/actor/ActorFragment.kt

```diff
@@ -1,29 +1,158 @@
+        )
         arguments?.let {
-            it.getInt(BUNDLE_ID_ACTOR_DETAIL)
+            idActor = it.getInt(BUNDLE_ID_ACTOR_DETAIL)
+        }
+        actorDetailPresenter.run {
+            idActor?.let {
+                setView(this@ActorFragment)
+                getActorDetail(it)
+                getExternal(it)
+                getListMovieOfActor(it)
+            }
+        }
+    }
+
+    override fun onEvent() {
+        imageBack.setOnClickListener {
+            requireActivity().supportFragmentManager.popBackStack()
+        }
+    }
+
+    override fun loadContentActorOnSuccess(actorDetail: ActorDetail) {
+        initDataActorDetail(actorDetail)
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun initDataActorDetail(actorDetail: ActorDetail) {
+        resources.apply {
+            actorDetail.run {
+                name?.let { textNameActor.text = it }
+                birthday?.let {
+                    textBirthday.text = formatHTML(getString(R.string.birthday), it)
+                }
+                address?.let {
+                    textAddress.text = formatHTML(getString(R.string.place_of_birth), it)
+                }
+                biography?.let {
+                    textBiography.text = formatHTML(getString(R.string.biography), it)
+                }
+                gender?.let {
+                    textGender.text =
+                        formatHTML(getString(R.string.gender), Gender.values()[it].name)
+                }
+                imageUrl?.let { imagePosterActor.loadFromUrl(Constant.BASE_URL_IMAGE + it) }
+            }
         }
     }
 
-    override fun loadContentActorOnSuccess(actorDetail: ActorDetail) {}
+    override fun loadContentExternalOnSuccess(external: External) {
+        external.run {
+            facebook?.let {
+                if (it != "")
+                    imageFacebook.toVisible()
+                else
+                    imageFacebook.toGone()
+            } ?: imageFacebook.toGone()
+            twitter?.let {
+                if (it != "")
+                    imageTwitter.toVisible()
+                else
+                    imageTwitter.toGone()
+            } ?: imageTwitter.toGone()
+            instagram?.let {
+                if (it != "")
+                    imageInstagram.toVisible()
+                else
+                    imageInstagram.toGone()
+            } ?: imageInstagram.toGone()
 
-    override fun loadContentExternalOnSuccess(external: External) {}
+            imageFacebook.setOnClickListener {
+                facebook?.let { it1 -> openExternal(Constant.URL_FACEBOOK + it1) }
+            }
+            imageTwitter.setOnClickListener {
+                twitter?.let { it1 -> openExternal(Constant.URL_TWITTER + it1) }
+            }
+            imageInstagram.setOnClickListener {
+                instagram?.let { it1 -> openExternal(Constant.URL_INSTAGRAM + it1) }
```

Comment:
Đặt tên cho có nghĩa. Không có định nghĩa tên theo số 1,2,3
```suggestion
            imageFacebook.setOnClickListener {
                facebook?.let { fb -> openExternal(Constant.URL_FACEBOOK + fb) }
            }
            imageTwitter.setOnClickListener {
                twitter?.let { twt-> openExternal(Constant.URL_TWITTER + twt) }
            }
            imageInstagram.setOnClickListener {
                instagram?.let { instag -> openExternal(Constant.URL_INSTAGRAM + instag) }
```